File: app/src/main/java/com/example/saka/myapplication/FragmentProfile.java
package com.example.saka.myapplication;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.example.saka.myapplication.adaper.OptionAdapter;
import com.example.saka.myapplication.model.myFirebase;
import com.google.firebase.auth.FirebaseAuth;

public class FragmentProfile extends Fragment implements View.OnClickListener, AdapterView.OnItemClickListener {

    private String[] options = {"New post", "Friend request", "Friends", "Settings", "About", "Log out"};
    private int[] icons = {R.drawable.ic_new_post, R.drawable.ic_add_circle, R.drawable.ic_friend,
            R.drawable.ic_settings, R.drawable.ic_about, R.drawable.ic_log_out};

    public FragmentProfile() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View rootView = inflater.inflate(R.layout.fragment_profile, container, false);
        RelativeLayout profileLayout = (RelativeLayout) rootView.findViewById(R.id.pf_bar);
        ListView listView = (ListView) rootView.findViewById(R.id.pf_list_view);
        TextView myName = (TextView) rootView.findViewById(R.id.pf_txt_name);
        TextView email = (TextView) rootView.findViewById(R.id.pf_txt_email);
        profileLayout.setOnClickListener(this);
        myName.setText(myFirebase.getUser(myFirebase.getMyUid()).getName());
        email.setText(FirebaseAuth.getInstance().getCurrentUser().getEmail());
        OptionAdapter adapter = new OptionAdapter(getContext(), options, icons);
        listView.setDivider(null);
        listView.setAdapter(adapter);

        listView.setOnItemClickListener(this);
        return rootView;
    }

    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.pf_bar) {
            //// TODO: 05-Jun-17 create activity profile
        }
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
        switch (position) {
            case 0:
                startActivity(new Intent(getActivity(), NewPostActivity.class));
                break;
            case 1:
                startActivity(new Intent(getActivity(), AddFriendActivity.class));
                break;
            case 2:
                //startActivity(new Intent(getActivity(), FriendListActivity.class));
                break;
            case 5:
                FirebaseAuth.getInstance().signOut();
                startActivity(new Intent(getActivity(), LogInActivity.class));
                getActivity().finish();
                break;
        }
    }
}   ///////////////////////////////////////////////////////////////////////////////////////////////

File: app/src/main/java/com/example/saka/myapplication/FragmentProfile.kt
package com.example.saka.myapplication

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView
import com.example.saka.myapplication.adaper.OptionAdapter
import com.example.saka.myapplication.model.MyFirebase.getUser
import com.example.saka.myapplication.model.MyFirebase.myUid
import com.example.saka.myapplication.ui.AddFriendActivity
import com.example.saka.myapplication.ui.LogInActivity
import com.example.saka.myapplication.ui.NewPostActivity
import com.google.firebase.auth.FirebaseAuth

class FragmentProfile : Fragment(), View.OnClickListener, OnItemClickListener {

    private val options = arrayOf(
        "New post",
        "Friend request",
        "Friends",
        "Settings",
        "About",
        "Log out"
    )
    private val icons = intArrayOf(
        R.drawable.ic_new_post,
        R.drawable.ic_add_circle,
        R.drawable.ic_friend,
        R.drawable.ic_settings,
        R.drawable.ic_about,
        R.drawable.ic_log_out,
    )

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View? {
        // Inflate the layout for this fragment
        val rootView = inflater.inflate(R.layout.fragment_profile, container, false)
        val profileLayout = rootView.findViewById<View>(R.id.pf_bar) as RelativeLayout
        val listView = rootView.findViewById<RecyclerView>(R.id.pf_list_view)
        val myName = rootView.findViewById<View>(R.id.pf_txt_name) as TextView
        val email = rootView.findViewById<View>(R.id.pf_txt_email) as TextView
        profileLayout.setOnClickListener(this)
        myName.text = getUser(myUid).name
        email.text = FirebaseAuth.getInstance().currentUser!!.email
        val adapter = OptionAdapter(options, icons)
        listView.adapter = adapter
        return rootView
    }

    override fun onClick(v: View) {
        if (v.id == R.id.pf_bar) {
            // // TODO: 05-Jun-17 create activity profile
        }
    }

    override fun onItemClick(parent: AdapterView<*>?, view: View, position: Int, id: Long) {
        when (position) {
            0 -> startActivity(Intent(activity, NewPostActivity::class.java))
            1 -> startActivity(Intent(activity, AddFriendActivity::class.java))
            2 -> {}
            5 -> {
                FirebaseAuth.getInstance().signOut()
                startActivity(Intent(activity, LogInActivity::class.java))
                requireActivity().finish()
            }
        }
    }
} // /////////////////////////////////////////////////////////////////////////////////////////////
